app/javascript/utils/utils.js

```diff
@@ -32,7 +32,8 @@ export const filterChanged = (prev, current) => {
   const changedFields = {}
   forEach(current, (value, key) => {
     if (!isEqual(prev[key], value)) {
-      if (isPlainObject(value) && prev[key] !== null) {
+      // date fields need to be treated as whole
+      if (isPlainObject(value) && prev[key] !== null && !('year' in value)) {
```

Comment:
can you move ('year' in value) to a function like `isDateObject()` that checks if day/month/year are present and are the only fields present? I could see some other objects containing a `year` field without being a date object